system-addon/lib/UserDomainAffinityProvider.jsm

```diff
@@ -91,6 +89,13 @@ this.UserDomainAffinityProvider = class UserDomainAffinityProvider {
     }
   }
 
+  get history() {
+    if (!this._history) {
+      this._history = Cc["@mozilla.org/browser/nav-history-service;1"].getService(Ci.nsINavHistoryService);
```

Comment:
There's a `XPCOMUtils.defineLazyServiceGetter` although unit-entry just drops it so individual tests manually set a global matching the jsm's expected global.

We could fix up unit-entry to do something smarter.. but maybe later.. :p